sys/include/net/gnrc/tcp/hdr.h

```diff
@@ -29,24 +29,18 @@ extern "C" {
 #endif
 
 /**
- * @brief TCP Options could contain up to 10 32-Bit values of Information.
- */
-#define TCP_MAX_HDR_OPTIONS 10
-
-/**
  * @brief TCP header definition
  */
 typedef struct __attribute__((packed)) {
-    network_uint16_t src_port;                      /**< source port, in network byte order */
-    network_uint16_t dst_port;                      /**< destination port, in network byte order */
-    network_uint32_t seq_num;                       /**< sequence number, in network byte order */
-    network_uint32_t ack_num;                       /**< Acknowledgement number, in network byte order */
-    network_uint16_t off_ctl;                       /**< Data Offset and control Bits in network byte order */
-    network_uint16_t window;                        /**< window, in network byte order */
-    network_uint16_t checksum;                      /**< checksum, in network byte order */
-    network_uint16_t urgent_ptr;                    /**< urgent pointer, in network byte order */
-    network_uint32_t options[TCP_MAX_HDR_OPTIONS];  /**< Option Fields (Optional) */
-} tcp_hdr_t;
+    network_uint16_t src_port;    /**< source port, in network byte order */
+    network_uint16_t dst_port;    /**< destination port, in network byte order */
+    network_uint32_t seq_num;     /**< sequence number, in network byte order */
+    network_uint32_t ack_num;     /**< Acknowledgement number, in network byte order */
+    network_uint16_t off_ctl;     /**< Data Offset and control Bits in network byte order */
+    network_uint16_t window;      /**< window, in network byte order */
+    network_uint16_t checksum;    /**< checksum, in network byte order */
+    network_uint16_t urgent_ptr;  /**< urgent pointer, in network byte order */
+} gnrc_tcp_hdr_t;
```

Comment:
I'd rather leave it `tcp_hdr_t` and further move the header (similar to udp) to `sys/include/net/tcp.h`, as I already did in #6509. However, I agree to removing the options array.